test/driver.js

```diff
@@ -278,6 +276,83 @@ var rasterizeAnnotationLayer = (function rasterizeAnnotationLayerClosure() {
   return rasterizeAnnotationLayer;
 })();
 
+/**
+ * @class
+ */
+var rasterizeXfaLayer = (function rasterizeXfaLayerClosure() {
+  const SVG_NS = "http://www.w3.org/2000/svg";
+
+  const styles = {
+    common: {
+      file: "../web/xfa_layer_builder.css",
+      promise: null,
+    },
+  };
+
+  function getXfaLayerStyle() {
+    return loadStyles(styles);
+  }
+
+  // eslint-disable-next-line no-shadow
+  function rasterizeXfaLayer(ctx, viewport, xfa, fontRules) {
+    return new Promise(function (resolve, reject) {
+      // Building SVG with size of the viewport.
+      const svg = document.createElementNS(SVG_NS, "svg:svg");
+      svg.setAttribute("width", viewport.width + "px");
+      svg.setAttribute("height", viewport.height + "px");
+
+      // Adding element to host our HTML (style + text layer div).
+      const foreignObject = document.createElementNS(
+        SVG_NS,
+        "svg:foreignObject"
+      );
+      foreignObject.setAttribute("x", "0");
+      foreignObject.setAttribute("y", "0");
+      foreignObject.setAttribute("width", viewport.width + "px");
+      foreignObject.setAttribute("height", viewport.height + "px");
+      const style = document.createElement("style");
+      const stylePromise = getXfaLayerStyle();
+      foreignObject.appendChild(style);
+      const div = document.createElement("div");
+      foreignObject.appendChild(div);
+
+      stylePromise
+        .then(async cssRules => {
+          style.textContent = fontRules + "\n" + cssRules;
+          // console.log(style);window.alert();
+
+          pdfjsLib.XfaLayer.render({
+            xfa,
+            div,
+            viewport: viewport.clone({ dontFlip: true }),
+            intent: "print",
```

Comment:
This should be `display` instead, right?
Otherwise you're not actually testing the *exact* same code-paths as used when rendering in e.g. the viewer.
Furthermore, note that we also use a `print` flag in some reference tests, so having this hard-coded could also lead to inconsistencies elsewhere.